### ISSUE TITLE
Open CVE on vendored jquery

### ISSUE THREAD

**Issue by msarahan** [matplotlib/matplotlib#11199]
### Bug report

**Bug summary**

Your vendored jquery is old and has a known security vulnerability, https://nvd.nist.gov/vuln/detail/CVE-2016-7103

**Code for reproduction**

Look at https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/web_backend/jquery/js/jquery-1.11.3.js

**Humble request**

Please update your vendored jquery.

**Comment by anntzer** [comment]
Would it make sense to not vendor jquery in the repo, but
1) change setup.py so that it fetches jquery at build time, normally online but configurably (via e.g. environment variable) from a local path (to satisfy distro packager constraints).
2) the sdist would not contain jquery but follow the same idea
3) the wheels would contain jquery.

Right now jquery weights 1.3Mb; compare this with the baseline test images that weight ~50Kb each and that we are painstakingly trying to avoid adding to the repo.

**Comment by WeatherGod** [comment]
That might run afoul with the Debian packaging rules. But, certainly, there
has to be a better solution than vendoring a js library...

**Comment by anntzer** [comment]
Hence the "configurably (via e.g. environment variable) from a local path (to satisfy distro packager constraints)."

**Comment by WeatherGod** [comment]
Ah, I missed that. I wonder what jupyter is doing?

**Comment by anntzer** [comment]
Installing the notebook from a source checkout requires bower/npm/whateverjs: https://jupyter-notebook.readthedocs.io/en/latest/contributing.html https://github.com/jupyter/notebook/blob/master/setupbase.py#L565
I'm basically advocating for the same model, but replacing all the relevant tooling by "download jquery".

Actually now that I think of it, for debian & friends the even simpler solution is to also bundle jquery in the sdists (they're building from a sdist, not from a checkout).  See comment in linked jupyter setupbase:
```
                # If we're an sdist, we aren't a repo and everything should be present.
                # Don't rebuild js/css in that case.
```

**Comment by dstansby** [comment]
Do we actually need the non-minimal version of jquery? It doesn't seem to be referenced anywhere else.

**Comment by anntzer** [comment]
@tacaswell told me it's the debian packaging rules...

**Comment by QuLogic** [comment]
In Fedora, all sources should be in its preferred form (i.e., editable by someone without going through hoops). That means non-minified sources. Ideally, the minified form should be generated from that source as well during package builds, but that's not always the case.

**Comment by anntzer** [comment]
(I don't think #11246 is actually that hard to push to completion if anyone wants to take a stab at it...)

**Comment by tacaswell** [comment]
This is addressed (via #13445 ).
